### PR TITLE
Remove ProtocolData proposal

### PR DESCRIPTION
**Pull request by joe-bowman**
## 1. Summary
Fixes QCK-486

## 2.Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## 3. Implementation details

Add proposal for removing protocol data by key.

### CODE REVIEW

File: x/participationrewards/types/msgs.go [hunk 1]
```diff
@@ -20,6 +20,8 @@ const (
 var (
 	_ sdk.Msg            = &MsgSubmitClaim{}
 	_ legacytx.LegacyMsg = &MsgSubmitClaim{}
+
```
Comment (ajansari95) [inline x/participationrewards/types/msgs.go:23]:
```suggestion

```

**Review by ajansari95 — APPROVED** [review]
lgtm

**Review by muku314115 — APPROVED** [review]

File: x/participationrewards/types/msgs.go [hunk 2]
```diff
@@ -102,3 +104,22 @@ func (msg MsgSubmitClaim) ValidateBasic() error {
...
+}
+
+// GetSigners Implements Msg.
+func (msg MsgGovRemoveProtocolData) GetSigners() []sdk.AccAddress {
+	fromAddress, _ := sdk.AccAddressFromBech32(msg.Authority)
+	return []sdk.AccAddress{fromAddress}
+}
+
+// Validate.
+func (msg MsgGovRemoveProtocolData) ValidateBasic() error { return nil }
```
Comment (aljo242) [inline x/participationrewards/types/msgs.go:125]: Let's add address validation here for `authority`.

**Review by aljo242 — CHANGES_REQUESTED** [review]

File: x/participationrewards/types/msgs.go [hunk 3]
```diff
@@ -88,17 +92,60 @@ func (msg MsgSubmitClaim) ValidateBasic() error {
...
-	if len(errors) > 0 {
-		return multierror.New(errors)
+	if len(errs) > 0 {
+		return multierror.New(errs)
 	}
 
 	return nil
 }
+
+// NewMsgGovCloseChannel - construct a msg to update signalled intent.
```
Comment (joe-bowman) [inline x/participationrewards/types/msgs.go:110]:
```suggestion
// NewMsgGovRemoveProtocolData - construct a msg to update signalled intent.
```

Comment (joe-bowman) [reply x/participationrewards/types/msgs.go:110]: fix comment copypasta

File: x/participationrewards/types/msgs.go [hunk 4]
```diff
@@ -88,17 +92,60 @@ func (msg MsgSubmitClaim) ValidateBasic() error {
...
-	if len(errors) > 0 {
-		return multierror.New(errors)
+	if len(errs) > 0 {
+		return multierror.New(errs)
 	}
 
 	return nil
 }
+
+// NewMsgGovRemoveProtocolData - construct a msg to update signalled intent.
```
Comment (joe-bowman) [inline x/participationrewards/types/msgs.go:110]:
```suggestion
// NewMsgGovRemoveProtocolData - construct a governance proposal msg to remove protocoldata by key.
```

File: x/participationrewards/types/msgs_test.go [hunk 5]
```diff
@@ -205,3 +205,146 @@ func TestNewMsgSubmitClaim(t *testing.T) {
...
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			msg := types.MsgGovRemoveProtocolData{
+				Title:       tt.fields.Title,
+				Description: tt.fields.Description,
+				Key:         tt.fields.Key,
+				Authority:   tt.fields.Authority,
+			}
+			if got := msg.GetSigners(); !reflect.DeepEqual(got, tt.want) {
+				err := fmt.Errorf("MsgSubmitClaim.GetSigners() = %v, want %v", got, tt.want)
```
Comment (joe-bowman) [inline x/participationrewards/types/msgs_test.go:345]:
```suggestion
				err := fmt.Errorf("MsgGovRemoveProtocolData.GetSigners() = %v, want %v", got, tt.want)
```

**Review by ajansari95 — DISMISSED** [review]
lgtm

**Review by muku314115 — APPROVED** [review]
lgtm

**Review by ajansari95 — APPROVED** [review]
lgtm

**Pull request merged**